microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -693,6 +695,49 @@ def verify_dpdk_send_receive_gb_hugepages_netvsc(
         except UnsupportedPackageVersionException as err:
             raise SkippedException(err)
 
+    @TestCaseMetadata(
+        description=(
+            """
+                Run the L3 forwarding test for DPDK
+        """
+        ),
+        priority=4,
```

Comment:
Why set it to P4?

Reply:
My logic was:

- It's a test that is checking for packet forwarding performance regressions for network virtual appliances. 
- It requires lots of resources and will take a long time to run. 
- It will run on a very narrow set of images and doesn't need to be picked up in every pipeline.

Reply:
1. What's the impact, if the test case failed? P4 means it's rare case to happen, and if it happens, it's little harmful.
2. How long does it take? From the test case name, it should be a validation test case. If a few packages are forwarded successfully, it should pass. It doesn't need long time to run. If it's a perf or stress test case, I agree it may need longer time.
3. If some images doesn't support this test case, it should be skipped, instead of use the priority to control it.

Reply:
These are good points.

1. I see what you mean now, I was worrying about pipelines wasting time deploying 3 VMs just to skip the test. It's an important test if it fails and catches a bug. I'll pull to 3.
2. The setup takes a while. You have to build DPDK each time to run it, l3fwd is a sample app and I don't think it's normally part of the DPDK package. This is a tricky test case. It is both a validation case (we check for 0 or no throughput) and a perf case (we're using ntttcp, checking the perf numbers, and creating a perf message). It might make more sense to put it in the perf suite?
3. The test does skip unsupported images at runtime. I don't want to maintain this case on every target we normally support for the DPDK tests, it's too complex. We will add support for more kernels later if we have an ask. Is there a way to get LISA to skip deploying the environment if the image isn't Ubuntu 22.04 or above? I thought we didn't have a way to enforce that before runtime.

Reply:
1. P3 looks good to me.
2. For installation, make sure they are running in parallel, so install 3 nodes use similar time as 2 nodes.
3. It can be split to two cases, one is for functionality validation, another is for perf.
4. Distro detection cannot happen before a real deployment, but it should be ok to spend some time to deploy but not used. It's not too long in the whole progress.

Reply:
moved to pri3 and added to description

Reply:
I can't speed the DPDK build up, we are only building DPDK on one node: the forwarder.
Now that you mention it there might be some build optimizations I could apply to all the DPDK installations? I will look into that in another PR.

Reply:
@squirrelsc split perf and functional tests and removed perf message post in functional test.